public/app/features/explore/utils/links.ts

```diff
@@ -121,10 +121,29 @@ export const getFieldLinksForExplore = (options: {
         }
         return linkModel;
       } else {
+        //TODO some sort of global use of transformationData.mappings
+        const internalLinkSpecificVars: ScopedVars = {};
+        if (link.internal?.transformationData?.transformations) {
+          const fieldValue = field.values.get(rowIndex);
+          link.internal?.transformationData?.transformations.forEach((transformation) => {
+            if (transformation.type === 'regex' && transformation.expression) {
+              const regexp = new RegExp(transformation.expression, 'g');
```

Comment:
I think global selector will not work properly with capturing, e.g.:

```
fieldValue.match(/app=(\w+)/)
(2) ['app=foo', 'foo', index: 0, input: 'app=foo host=demo-001', groups: undefined]
fieldValue.match(/app=(\w+)/g)
['app=foo']
```

In this case we need to capture "foo", not all matches of entire "app=(\w+)"

Reply:
The global match was me thinking ahead on future plans where one correlation could match multiple times, like the (Batman|Superman) example. We can take out the global match for now.

Reply:
Let's keep the scope of the task. We'll tackle multiple/inline links in https://github.com/grafana/grafana/issues/62056. For now let's just focus on getting simple regex (no capturing) assigned to a single variable with provisioning (no UI changes). It won't be possible to easily test it with Logs Panel (we need this -> https://github.com/grafana/grafana/issues/61782), but it will be possible to test it with Table visualizations.